### PR TITLE
Add Ruff, Djlint and v5 branch changes

### PR DESCRIPTION
**Pull request by alexandreNoens**
## This Pull Request is not intended to be merged

When moving from `v2` to `v3`, the use of **Docker** was added. 
However, for certain projects, we do not need Docker. Therefore, we created a `v3-legacy` branch that does not use Docker but includes the updates made to the branch which continued with the use of Docker (MakeCitron/v5).

### CODE REVIEW

File: MakeCitron.Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 include config.Makefile
 -include config.custom.Makefile
 
-BASEVERSION ?= v2
+BASEVERSION ?= v3
```
Comment (glepretre) [inline MakeCitron.Makefile:4]: Will conflict with already existing v3 branch? I expected something like `v3-legacy`

File: base.Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 VERSION := 1.4.16
 
 # Use bash
-SHELL := /bin/bash
+SHELL := bash
```
Comment (glepretre) [inline base.Makefile:4]: Why do we need to simplify? Does not seem very UNIX to me.

File: dots/pyproject.toml [hunk 3]
```diff
@@ -1,26 +1,7 @@
...
-[tool.isort]
-multi_line_output = 3
-default_section = "THIRDPARTY"
-include_trailing_comma = true
-force_grid_wrap = 0
-combine_as_imports = true
+[tool.ruff.format]
+
+[tool.ruff.lint.isort]
+combine-as-imports = true
```
Comment (glepretre) [inline dots/pyproject.toml]: Would be nice to fix the new line at the end of file.

File: MakeCitron.Makefile (hunk 1 above)
Comment (glepretre) [reply MakeCitron.Makefile:4]: Mob review: it's just an example.

File: base.Makefile (hunk 2 above)
Comment (glepretre) [reply base.Makefile:4]: mob: Changed between v2 and v3, not related to this branch.

**Review by glepretre — COMMENTED** [review]
Mob: To review please @gentooboontoo

**Comment by glepretre** [comment]
Mob : ping @gentooboontoo

File: base.Makefile [hunk 4]
```diff
@@ -310,7 +323,7 @@ clea%: least-specific-clean ## clean: Clean all built assets
 #
 lint-pytho%: ## lint-python: Lint python source
 	$(LOG)
-	$(PYTEST) --flake8 --isort -m "flake8 or isort" "$(PYTHON_SRCDIR)" --ignore=lib/frontend/static
+	$(RUFF) check $(LINTED_PYTHON_DIRS)
```
Comment (gentooboontoo) [inline base.Makefile:327]: djLint-related lines are missing from backport (`TEMPLATE_DIRS` variable too).

**Review by gentooboontoo — CHANGES_REQUESTED** [review]

**Review by gentooboontoo — APPROVED** [review]

**Comment by gentooboontoo** [comment]
Rappel probablement inutile mais à clôturer sans merger une fois les revues terminées.

**Pull request closed without merging**